src/@utils/order.ts

```diff
@@ -182,60 +231,66 @@ export async function handleComputeOrder(
     '[compute] Handle compute order for asset type: ',
     asset.metadata.type
   )
+  LoggerInstance.log('[compute] Using initializeData: ', initializeData)
 
-  if (
-    initializeData.providerFee &&
-    initializeData.providerFee.providerFeeAmount !== '0'
-  ) {
-    const baseToken =
-      asset?.accessDetails?.type === 'free'
-        ? getOceanConfig(asset.chainId).oceanTokenAddress
-        : asset?.accessDetails?.baseToken?.address
-    const txApproveWei = await approveWei(
-      web3,
-      accountId,
-      baseToken,
-      asset?.accessDetails?.datatoken?.address,
-      initializeData?.providerFee?.providerFeeAmount
-    )
-    if (!txApproveWei) {
-      toast.error('Failed to approve provider fees!')
-      return
+  try {
+    // Return early when valid order is found, and no provider fees
+    // are to be paid
+    if (
+      initializeData?.validOrder &&
+      (!initializeData.providerFee ||
+        initializeData?.providerFee?.providerFeeAmount === '0')
```

Comment:
think we do not need this check here `initializeData?.providerFee?.providerFeeAmount === '0')`, because for the algo for eg if the order is valid and you have to pay the provider fees those will show up in the dataset field and the providerFee?.providerFeeAmount for the algorithm will be '0'. Posting an example below for wich we should reuse orders on both datasets and algorithm:
`{
  "algorithm": {
    "datatoken": "0x37E7aeC8f9056ACf95C893a9B883c6b41e7651a8",
    "providerFee": {
      "providerData": "0x7b22656e7669726f6e6d656e74223a22636f6d707574652d76342d656e67696e652d6465762d656e7632227d",
      "providerFeeAddress": "0x00c6A0BC5cD0078d6Cd0b659E8061B404cfa5704",
      "providerFeeAmount": "0",
      "providerFeeToken": "0x5e8DCB2AfA23844bcc311B00Ad1A0C30025aADE9",
      "r": "0x502634912c556526f842d4092b7811ccea873ea0e509bfe8a723a4e0b6317aeb",
      "s": "0x72536c7586fe132d7f76377f897c85a8a245a4cf00fecf2aa26531b8ee7cba94",
      "v": 27,
      "validUntil": 1655980103
    },
    "validOrder": "0x2f17111b6d6e509c98afbf2740510b2789e4c20c3f1ea7b53a9c3146447504cc"
  },
  "datasets": [
    {
      "datatoken": "0x497E4bF570e8cB61D080a891D0C644D1d3001fb2",
      "providerFee": {
        "providerData": "0x7b22656e7669726f6e6d656e74223a22636f6d707574652d76342d656e67696e652d6465762d656e7632227d",
        "providerFeeAddress": "0x00c6A0BC5cD0078d6Cd0b659E8061B404cfa5704",
        "providerFeeAmount": "1983333333333333400",
        "providerFeeToken": "0x5e8DCB2AfA23844bcc311B00Ad1A0C30025aADE9",
        "r": "0x3c76a4970963ca1b37e254da23e10969a034094a1a884fccd7a91e0b70e01c47",
        "s": "0x54c274035b5531cd8615b679642e0c90bc44893226026579229861c86f47e843",
        "v": 27,
        "validUntil": 1655980103
      },
      "validOrder": "0x0248852caa92e34a9deae4607cd12ba030c39d9830eb8348d45e99fe646da7bd"
    }
  ]
}`

Reply:
cool, thanks, reverted that change. But in general it's ok to return early with existing valid order? Cause wasn't sure under which circumstances `!initializeData.providerFee` is true.

Reply:
yes should be fine, but just realised a possible issue after the changes made on provider yesterday for that fix start compute job fix.